core/config/project_settings.cpp

```diff
@@ -48,11 +48,22 @@ ProjectSettings *ProjectSettings::get_singleton() {
 	return singleton;
 }
 
+String ProjectSettings::get_project_data_dir_name() const {
+	return ".godot";
+}
+
+String ProjectSettings::get_project_data_path() const {
+	String project_data_dir_name = get_project_data_dir_name();
+	return "res://" + project_data_dir_name;
+}
+
 String ProjectSettings::get_resource_path() const {
 	return resource_path;
 }
 
-const String ProjectSettings::IMPORTED_FILES_PATH("res://.godot/imported");
+String ProjectSettings::get_imported_files_path() const {
+	return get_project_data_path() + "/imported";
```

Comment:
```suggestion
	return get_project_data_path().plus_file("imported");
```

This will ensure that only one `/` is added, even if the `get_project_data_path()` ends with a `/`.